src/Workspaces/Core/Portable/Shared/Extensions/ChildSyntaxListExtensions.cs

```diff
@@ -0,0 +1,34 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Linq;
+
+namespace Microsoft.CodeAnalysis.Shared.Extensions
+{
+    internal static class ChildSyntaxListExtensions
+    {
+        public static SyntaxNodeOrToken First(this ChildSyntaxList childSyntaxList, Func<SyntaxNodeOrToken, bool> predicate)
+        {
+            foreach (var syntaxNodeOrToken in childSyntaxList)
```

Comment:
why not just use enumerable? to reduce allocation?

Reply:
> to reduce allocation?

Correct. The formatter is a huge source of allocations on a performance hot path.